cpp/src/search/search.cu

```diff
@@ -134,17 +134,20 @@ struct contains_scalar_dispatch {
     auto s = static_cast<const ScalarType *>(&value);
 
     if (col.has_nulls()) {
-      auto element_valid_pair = make_pair_iterator<Element, true>(*d_col);
+      auto element_valid_iter = make_pair_iterator<Element, true>(*d_col);
```

Comment:
I realized something that bothers me with make_pair_iterator -- there's no simple way to get one from an equivalent of `begin()` and `end()` on a column_device_view.  https://github.com/rapidsai/cudf/blob/da78798025c5a02472b4612e3a82266b27d3bc9e/cpp/include/cudf/column/column_device_view.cuh#L291-L301

Would it be possible to overload `begin()` and `end()` so they return a pair_iterator when the column is nullable? Or add `column_device_view::pair_begin()` and `pair_end()` for nullable columns? That would simplify this code a lot:

```
auto found_iter = thrust::find(rmm::exec_policy(stream)->on(stream),
                                      d_col->pair_begin(),
                                      d_col->pair_end(),
                                      val);
return found_iter != d_col->pair_end();
```

CC @jrhemstad @karthikeyann

Reply:
I could make this change, if we think it's a good idea.  I'd be inclined to add pair_begin() and pair_end() rather than overload begin() and end(), as it makes the distinction clearer.  However, since begin() and end() behavior is incorrect if the column has nulls, perhaps overloading begin() and end() is reasonable.

Reply:
But perhaps this would be best addressed in #3490, since it is still open.

Reply:
`pair_begin` and `pair_end` sound reasonable to me.

Reply:
@harrism @jrhemstad Shall I add this as part of pair iterator PR?

Reply:
Yes.